@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -648,17 +648,21 @@ export class VisibleTileSet {
     private markDataSourceTilesDirty(renderListEntry: DataSourceTileList) {
         const dataSourceCache = this.m_dataSourceCache.get(renderListEntry.dataSource.name);
         const retainedTiles: Set<number> = new Set();
-        renderListEntry.visibleTiles.forEach(tile => {
-            const tileCode = TileOffsetUtils.getKeyForTileKeyAndOffset(tile.tileKey, tile.offset);
-            retainedTiles.add(tileCode);
-            tile.reload();
-        });
-        renderListEntry.renderedTiles.forEach(tile => {
+
+        function markTileDirty(tile: Tile, tileGeometryManager: TileGeometryManager) {
             const tileCode = TileOffsetUtils.getKeyForTileKeyAndOffset(tile.tileKey, tile.offset);
             if (!retainedTiles.has(tileCode)) {
                 retainedTiles.add(tileCode);
+                tileGeometryManager.initTile(tile);
```

Comment:
Would it be worth making a new method to `reinit` a tile, this allocates a new loader, however the internal state should just need to be cleared, @StefanDachwitz, do you have an opinion?

Reply:
I guess I would implement a reinit(), to clear() the tile of any old geometry, and to make sure the old loader is not adding anything, is cancelled and destroyed properly.